src/metrics/classifier.ts

```diff
@@ -1,15 +1,77 @@
-import { Tensor, equal, sum, div } from '@tensorflow/tfjs-core';
+import { Tensor, equal, sum, div, math, Tensor1D, divNoNan, concat, mul, add, cast } from '@tensorflow/tfjs-core';
 import { checkArray } from '../utils/validation';
+import { getDiagElements } from '../linalg/utils';
+import { LabelEncoder } from '../preprocess';
 
-export const accuracyScore = ( yTrue: Tensor | string[] | number[], yPred: Tensor | string[] | number[] ): number => {
+export type ClassificationReport = {
+  precisions: Tensor;
+  recalls: Tensor;
+  f1s: Tensor;
+  confusionMatrix: Tensor;
+  classes: Tensor;
+  accuracy: number;
+  averagePrecision: number;
+  averageRecall: number;
+  averageF1: number;
+};
+
+export type ClassificationAverageTypes = 'macro' | 'weighted'
+
+/**
+ * Check that if yTrue and yPred are of same length
+ * @param yTrue true labels
+ * @param yPred predicted labels
+ */
+export const checkSameLength = (yTrue: Tensor | string[] | number[], yPred: Tensor | string[] | number[]): [ Tensor1D, Tensor1D, number ] => {
   const yTrueTensor = checkArray(yTrue, 'any', 1);
   const yPredTensor = checkArray(yPred, 'any', 1);
   const yTrueCount = yTrueTensor.shape[0];
   const yPredCount = yPredTensor.shape[0];
   if (yTrueCount != yPredCount) {
     throw new Error('Shape of yTrue should match shape of yPred');
   }
+  return [ yTrueTensor as Tensor1D, yPredTensor as Tensor1D, yTrueCount ];
+};
+
+export const accuracyScore = (yTrue: Tensor | string[] | number[], yPred: Tensor | string[] | number[]): number => {
+  const [ yTrueTensor, yPredTensor, nLabels ] = checkSameLength(yTrue, yPred);
   // TODO(sugarspectre): Accuaracy score computation
-  const score = div(sum(equal(yPred, yTrue)), yTrueCount).dataSync()[0];
+  const score = div(sum(equal(yPredTensor, yTrueTensor)), nLabels).dataSync()[0];
   return score;
 };
+
+/**
+ * Generate classification report
+ * @param yTrue true labels
+ * @param yPred predicted labels
+ * @returns classification report object, the struct of report will be like following
+ */
+export const classificationReport = async(yTrue: Tensor | string[] | number[], yPred: Tensor | string[] | number[], average: ClassificationAverageTypes = 'weighted'): Promise<ClassificationReport> => {
```

Comment:
```suggestion
export const generateClassificationReport = async(yTrue: Tensor | string[] | number[], yPred: Tensor | string[] | number[], average: ClassificationAverageTypes = 'weighted'): Promise<ClassificationReport> => {
```

A function must be a verb namely action.

Reply:
Or implement as a class because you have defined the type `ClassificationReport`.